flowmachine/flowmachine/features/subscriber/home_location_monthly.py

```diff
@@ -0,0 +1,197 @@
+        self.home_last_month = home_this_month
+        self.modal_lookback = modal_lookback
+
+        self.active_subs = ActiveSubscribers(
+            start_date=self.window_start,
+            end_date=self.window_stop,
+            active_days=active_days,
+            interval=interval,
+            events_tables=events_tables,
+        )
+
+        if self._window_start >= self._window_stop - timedelta(days=modal_lookback):
+            full_range_start = self._window_stop - timedelta(days=modal_lookback)
+        else:
+            full_range_start = self._window_start
+        full_range_stop = self._window_stop
+
+        all_locations = {
+            day: LastLocation(
+                start=day.strftime("%Y-%m-%d"),
+                stop=(day + timedelta(days=1)).strftime("%Y-%m-%d"),
+                spatial_unit=self.agg_unit,
+                subscriber_subset=self.active_subs,
+                table=events_tables,
+            )
+            for day in rrule(DAILY, dtstart=full_range_start, until=full_range_stop)
+        }
+        modal_days = [
+            query
+            for (day, query) in all_locations.items()
+            if full_range_start <= day <= full_range_stop
+        ]
+        self.last_locations = [
+            query
+            for (day, query) in all_locations.items()
+            if self._window_start <= day <= self._window_stop
+        ]
+        self.modal_locations = ModalLocation(*modal_days)
+
+        # self.daily_location_frequency = LocationVisits(
+        #     self.last_locations  # Ask Jono about this
+        # )
+
+        super().__init__()
+
+    @property
+    def column_names(self) -> List[str]:
+        cols = ["subscriber", "location", "month"]
+        return cols
+
+    @property
+    def window_start(self):
+        return self._window_start.strftime("%Y-%m-%d")
+
+    @window_start.setter
+    def window_start(self, value):
+        if type(value) is str:
+            self._window_start = datetime.strptime(value, "%Y-%m-%d")
+        elif type(value) is datetime:
+            self._window_start = value
+        else:
+            raise TypeError("window_start must be datetime or yyyy-mm-dd")
+
+    @property
+    def window_stop(self):
+        return self._window_stop.strftime("%Y-%m-%d")
+
+    @window_stop.setter
+    def window_stop(self, value):
+        if type(value) is str:
+            self._window_stop = datetime.strptime(value, "%Y-%m-%d")
+        elif type(value) is datetime:
+            self._window_stop = value
+        else:
+            raise TypeError("window_stop must be datetime or yyyy-mm-dd")
+
+    def _make_query(self):
+
+        last_locations_clause = ""
+
+        for last_location in self.last_locations:
+            last_locations_clause += f"""
+SELECT subscriber, pcod, '{last_location.start}' AS day
+FROM ({last_location.get_query()}) AS tbl
+UNION ALL
+"""
+
+        last_locations_clause = last_locations_clause.rstrip("UNION ALL\n")
```

Comment:
Use a reduce and `.union(all=True)`, or `join` on the list? Or use `MultiLocation` which includes the date bit + handles non-admin levels.